src/main/java/com/yelp/nrtsearch/server/luceneserver/NRTReplicaNode.java

```diff
@@ -121,7 +123,9 @@ public synchronized void start(long primaryGen) throws IOException {
       // Swap in a SearcherManager that filters incompatible segment readers during refresh.
       // Updating the reference is not thread safe, but since this happens under the object lock
       // and before the shard has stared, nothing should access the manager before the swap.
+      ReferenceManager<IndexSearcher> oldMgr = mgr;
       mgr = new FilteringSegmentInfosSearcherManager(getDirectory(), this, mgr, searcherFactory);
```

Comment:
It appears as if `FilteringSegmentInfosSearcherManager` is using the old reference manager because it's passed in the constructor, but it only gets the current SegmentInfos: `((SegmentInfosSearcherManager) mgr).getCurrentInfos()`
Can we change the constructor to accept SegmentInfos and then pass it from here so that it's clear that the old reference manager is not accessed any more?

Reply:
Unfortunately, `SegmentInfosSearcherManager` is package private and cannot be used from `NRTReplicaNode` unless we relocate the class.

Reply:
Makes sense 👍